lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_methods.rb

```diff
@@ -155,7 +155,7 @@ def self.create_provision_request(*args)
       # Need to add the username into the array of params
       # TODO: This code should pass a real username, similar to how the web-service
       #      passes the name of the user that logged into the web-service.
-      args << User.lookup_by_userid("admin")
+      args.insert(1, User.lookup_by_userid("admin"))
```

Comment:
NOTE this matches how this was done prior to the ruby3 work but dropping the `if Array` check: https://github.com/ManageIQ/manageiq-automation_engine/pull/509/commits/dca9b9762033871966a04f9c676f6520aebab4a7#diff-8500593c270b3104728814f6355956c80683a3b17b15368f0d813d25b10a836cL147